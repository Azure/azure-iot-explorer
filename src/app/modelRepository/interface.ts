import { ModelRepositoryStateInterface } from "../shared/modelRepository/state";
import { StringMap } from '../api/models/stringMap';

export interface ModelRepositoryFormOpsInterface {
    setDirtyFlag: (dirty: boolean) => void;
    setRepositoryLocationSettings: (settings: ModelRepositoryStateInterface) => void;
    setRepositoryLocationSettingsErrors: (errors: StringMap<string> ) => void;
}

export const getInitialModelRepositoryFormOps = (): ModelRepositoryFormOpsInterface => ({
    setDirtyFlag: () => undefined,
    setRepositoryLocationSettings:  () => undefined,
    setRepositoryLocationSettingsErrors:  () => undefined
});