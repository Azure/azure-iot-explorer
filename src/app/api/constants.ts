/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
export enum HTTP_OPERATION_TYPES {
    Delete = 'DELETE',
    Get = 'GET',
    Patch = 'PATCH',
    Post = 'POST',
    Put = 'PUT'
}

export const APPLICATION_JSON = 'application/json';
export const ERROR_TYPES = {
    HTTP: 'http'
};
