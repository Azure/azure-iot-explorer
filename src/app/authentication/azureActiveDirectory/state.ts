/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import { IotHubDescription } from '../../api/models/iotHubDescription';
import { AzureSubscription } from '../../api/models/azureSubscription';
export interface AzureActiveDirectoryStateInterface {
    formState: 'initialized' | 'working' | 'failed' | 'idle' | 'keyPicked';
    iotHubKey: string;
    iotHubs: IotHubDescription[];
    subscriptions: AzureSubscription[];
    token: string;
}

export const getInitialAzureActiveDirectoryState = (): AzureActiveDirectoryStateInterface => ({
    formState: 'initialized',
    iotHubKey: undefined,
    iotHubs: [],
    subscriptions: [],
    token: undefined
});
