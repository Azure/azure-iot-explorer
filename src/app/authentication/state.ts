/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
export enum AuthenticationMethodPreference {
    AzureAD = 'AzureAD',
    ConnectionString = 'ConnectionString'
}

export interface AuthenticationStateInterface {
    formState: 'initialized' | 'working' | 'failed' | 'idle';
    preference: string;
}

export const getInitialAuthenticationState = (): AuthenticationStateInterface => ({
    formState: 'initialized',
    preference: undefined
});
