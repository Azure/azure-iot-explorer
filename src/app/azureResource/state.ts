/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import { AzureResource } from './models/azureResource';

export interface AzureResourceStateInterface {
    activeAzureResource?: AzureResource;
}

export const azureResourceStateInitial = (): AzureResourceStateInterface => {
    return {};
};
