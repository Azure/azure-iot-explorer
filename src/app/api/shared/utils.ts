/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import { createHmac } from 'crypto';
import { IoTHubConnectionSettings } from '../services/devicesService';
import { LIST_PLUG_AND_PLAY_DEVICES, SAS_EXPIRES_MINUTES } from '../../constants/devices';
import DeviceQuery, { QueryClause, ParameterType, OperationType } from '../models/deviceQuery';
import { RepoConnectionSettings } from '../services/digitalTwinsModelService';
import { AppEnvironment } from '../../constants/shared';

const MILLISECONDS_PER_SECOND = 1000;
const SECONDS_PER_MINUTE = 60;

export const enum PnPQueryPrefix {
    HAS_CAPABILITY_MODEL = 'HAS_CAPABILITYMODEL',
    HAS_INTERFACE = 'HAS_INTERFACE'
}

export interface GenerateSasTokenParameters {
    /**
     * URI prefix (by segment) of the endpoints that can be accessed with this token, starting with host name of the IoT hub (no protocol). For example, myHub.azure-devices.net/devices/device1
     */
    resourceUri: string;

    /**
     * The key to use for the token.
     */
    key: string;

    /**
     * Expiration time in minutes. Defaults to 5.
     */
    expiration?: number;

    /**
     * The name of the key being used. Omit if the token refers to device-registry credentials.
     */
    keyName?: string;
}

export const generateSasToken = (parameters: GenerateSasTokenParameters) => {
    let token = null;
    if (!!parameters.resourceUri && !!parameters.key) {
        const encodedUri = encodeURIComponent(parameters.resourceUri);

        const expires = Math.ceil((Date.now() / MILLISECONDS_PER_SECOND) + (parameters.expiration || SAS_EXPIRES_MINUTES) * SECONDS_PER_MINUTE);
        const toSign = encodedUri + '\n' + expires;

        const hmac = createHmac('sha256', new Buffer(parameters.key, 'base64'));
        hmac.update(toSign);
        const base64UriEncoded = encodeURIComponent(hmac.digest('base64'));

        token = `SharedAccessSignature sr=${encodedUri}&sig=${base64UriEncoded}&se=${expires}${parameters.keyName ? '&skn=' + parameters.keyName : ''}`;
    }
    return token;
};

export const generatePnpSasToken = (repositoryId: string, audience: string, secret: string, keyName: string) => {
    const now = new Date();
    const ms = 1000;
    const expiry = (now.setDate(now.getDate() + 1) / ms).toFixed(0);
    const encodedServiceEndpoint = encodeURIComponent(audience);
    const encodedRepoId = encodeURIComponent(repositoryId);
    const signature = [encodedRepoId, encodedServiceEndpoint, expiry].join('\n').toLowerCase();
    const sigUTF8 = new Buffer(signature, 'utf8');
    const secret64bit = new Buffer(secret, 'base64');
    const hmac = createHmac('sha256', secret64bit);
    hmac.update(sigUTF8);
    const hash = encodeURIComponent(hmac.digest('base64'));
    return `SharedAccessSignature sr=${encodedServiceEndpoint}&sig=${hash}&se=${expiry}&skn=${keyName}&rid=${repositoryId}`;
};

export const getRepoConnectionInfoFromConnectionString = (connectionString: string): RepoConnectionSettings => {
    const connectionObject: RepoConnectionSettings = {};
    connectionString.split(';')
    .forEach((segment: string) => {
        const keyValue = segment.split('=');
        switch (keyValue[0]) {
            case 'HostName':
            connectionObject.hostName = keyValue[1];
            break;
            case 'SharedAccessKeyName':
            connectionObject.sharedAccessKeyName = keyValue[1];
            break;
            case 'SharedAccessKey':
            connectionObject.sharedAccessKey = keyValue[1];
            break;
            case 'RepositoryId':
            connectionObject.repositoryId = keyValue[1];
            break;
            default:
            // we don't use other parts of connection string
            break;
        }
    });

    return connectionObject;
};

export const getConnectionInfoFromConnectionString = (connectionString: string): IoTHubConnectionSettings => {
    const connectionObject: IoTHubConnectionSettings = {};
    connectionString.split(';')
    .forEach((segment: string) => {
        const keyValue = segment.split('=');
        switch (keyValue[0]) {
            case 'HostName':
            connectionObject.hostName = keyValue[1];
            break;
            case 'SharedAccessKeyName':
            connectionObject.sharedAccessKeyName = keyValue[1];
            break;
            case 'SharedAccessKey':
            connectionObject.sharedAccessKey = keyValue[1];
            break;
            default:
            // we don't use other parts of connection string
            break;
        }
    });

    return connectionObject;
};

export const buildQueryString = (query: DeviceQuery) => {
    return query ?
        `${LIST_PLUG_AND_PLAY_DEVICES} ${queryToString(query)}` :
        LIST_PLUG_AND_PLAY_DEVICES;
};

export const queryToString = (query: DeviceQuery) => {
    if (query.deviceId) {
        return `WHERE STARTSWITH(devices.deviceId, '${query.deviceId}')`;
    }
    const clauseQuery = clauseListToString(query.clauses && query.clauses.filter(clause => !!clause.parameterType));
    if ('' !== clauseQuery) {
        return `WHERE (${clauseQuery })`;
    }
    return '';
};

export const clauseListToString = (clauses: QueryClause[]) => {
    return clauses && clauses.map(clause => clauseToString(clause)).join(' AND ') || '';
};

export const clauseToString = (clause: QueryClause) => {
    switch (clause.parameterType) {
        case ParameterType.capabilityModelId:
            return toPnPClause(PnPQueryPrefix.HAS_CAPABILITY_MODEL, clause.value);
        case ParameterType.interfaceId:
            return toPnPClause(PnPQueryPrefix.HAS_INTERFACE, clause.value);
        case ParameterType.edge:
            return toEdgeClause(clause.parameterType, clause.value);
        case ParameterType.status:
            return toDeviceStatusClause(clause.parameterType, clause.value);
        default:
            return clauseItemToString(clause.parameterType, clause.operation, clause.value);
    }
};

export const clauseItemToString = (fieldName: string, operation: OperationType, value: unknown) => {
    if (!fieldName || !operation || !value) {
        return;
    }
    return `${fieldName} ${operation} ${escapeValue(value as string)}`;
};

export const escapeValue = (value: string) => {
    const isString = new RegExp(/[^0-9.]/);
    const hasQuotes = new RegExp(/^'.*'$/);
    if (value.match(isString) && !value.match(hasQuotes)) {
        return `'${value}'`;
    }
    return value;
};

export const toPnPClause = (pnpFunctionName: string, value: string): string => {
    const urnVersionRegEx = new RegExp(/:[0-9]+$/);
    if (urnVersionRegEx.test(value)) {
        const urnVersion = urnVersionRegEx.exec(value)[0].replace(':', '');
        const urnName = value.replace(urnVersionRegEx, '');

        return `${pnpFunctionName}(${escapeValue(urnName)}, ${urnVersion})`;
    }
    return `${pnpFunctionName}('${value}')`; // when provided value is not urn with version, pass it in as string
};

export const toEdgeClause = (edgeFunctionName: string, value: string): string => {
    return `${edgeFunctionName}=${value === 'true' ? true : false}`;
};

export const toDeviceStatusClause = (edgeFunctionName: string, value: string): string => {
    return `${edgeFunctionName}='${value}'`;
};

export const isAppInElectron = () => {
    return process.env.NODE_ENV === AppEnvironment.DevelopmentElectron || process.env.NODE_ENV === AppEnvironment.ProdElectron;
};

export const isAppInDevelopmentMode = () => {
    return process.env.NODE_ENV === AppEnvironment.DevelopmentElectron || process.env.NODE_ENV === AppEnvironment.DevelopmentHosted;
};
