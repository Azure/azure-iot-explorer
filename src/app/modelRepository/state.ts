import { ModelRepositoryStateInterface } from "../shared/modelRepository/state";
import { StringMap } from '../api/models/stringMap';
import { REPOSITORY_LOCATION_TYPE } from "../constants/repositoryLocationTypes";

export interface ModelRepositoryFormStateInterface {
    dirty: boolean;
    repositoryLocationSettings: ModelRepositoryStateInterface;
    repositoryLocationSettingsErrors: StringMap<string>;
}

export const getInitialModelRepositoryFormState = (): ModelRepositoryFormStateInterface => ({
    dirty: false,
    repositoryLocationSettings: [{repositoryLocationType: REPOSITORY_LOCATION_TYPE.Public, value: '' }],
    repositoryLocationSettingsErrors: {}
});