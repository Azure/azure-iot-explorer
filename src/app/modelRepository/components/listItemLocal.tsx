/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { DefaultButton, PrimaryButton, Dialog, DialogFooter, TextField, Link } from '@fluentui/react';
import { REPOSITORY_LOCATION_TYPE } from '../../constants/repositoryLocationTypes';
import { ResourceKeys } from '../../../localization/resourceKeys';
import { NAVIGATE_BACK, FOLDER } from '../../constants/iconNames';
import { fetchDirectories } from '../../api/services/localRepoService';
import { getRootFolder, getParentFolder } from '../../shared/utils/utils';
import { ModelRepositoryConfiguration } from '../../shared/modelRepository/state';
import { ModelRepositoryFormType } from '../hooks/useModelRepositoryForm';
import { ListItemLocalLabel } from './listItemLocalLabel';

export interface ListItemLocalProps {
    index: number;
    item: ModelRepositoryConfiguration;
    formState: ModelRepositoryFormType;
    repoType: REPOSITORY_LOCATION_TYPE;
}

// tslint:disable-next-line: cyclomatic-complexity
export const ListItemLocal: React.FC<ListItemLocalProps> = ({ item, index, repoType, formState }) => {
    const { t } = useTranslation();
    const [{repositoryLocationSettings, repositoryLocationSettingsErrors }, {setRepositoryLocationSettings, setDirtyFlag}] = formState;
    const errorKey = repositoryLocationSettingsErrors[item.repositoryLocationType];

    let initialCurrentFolder = '';
    if (item && item.repositoryLocationType === repoType) {
        initialCurrentFolder = item.value || getRootFolder();
    }

    const [ subFolders, setSubFolders ] = React.useState([]);
    const [ currentFolder, setCurrentFolder ] = React.useState(initialCurrentFolder);
    const [ showError, setShowError ] = React.useState<boolean>(false);
    const [ showFolderPicker, setShowFolderPicker ] = React.useState<boolean>(false);

    React.useEffect(() => {
        setCurrentFolder(initialCurrentFolder);
    }, [initialCurrentFolder]); // tslint:disable-line: align

    const onChangeRepositoryLocationSettingValue = (value: string) => {
        const updatedRepositoryLocationSettings = repositoryLocationSettings.map((setting, i) => {
            if (i === index) {
                const updatedSetting = {...setting};
                updatedSetting.value = value;
                return updatedSetting;
            } else {
                return setting;
            }
        });

        setDirtyFlag(true);
        setRepositoryLocationSettings(updatedRepositoryLocationSettings);
    };

    const onFolderPathChange = (event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, newValue?: string) => {
        setCurrentFolder(newValue);
        onChangeRepositoryLocationSettingValue(newValue);
   };

    const onShowFolderPicker = () => {
        fetchSubDirectoriesInfo(currentFolder);
        setShowFolderPicker(true);
    };

    const dismissFolderPicker = () => {
        setCurrentFolder(item.value || getRootFolder());
        setShowFolderPicker(false);
    };

    const onSelectFolder = () => {
        onChangeRepositoryLocationSettingValue(currentFolder);
        setShowFolderPicker(false);
    };

    const onClickFolderName = (folder: string) => () => {
        const newDir = currentFolder ? `${currentFolder.replace(/\/$/, '')}/${folder}` : folder;
        setCurrentFolder(newDir);
        fetchSubDirectoriesInfo(newDir);
    };

    const onNavigateBack = () => {
        const parentFolder = getParentFolder(currentFolder);
        setCurrentFolder(parentFolder);
        fetchSubDirectoriesInfo(parentFolder);
    };

    const fetchSubDirectoriesInfo = (folderName: string) => {
        fetchDirectories(folderName).then(result => {
            setShowError(false);
            setSubFolders(result);
        }).catch(error => {
            setShowError(true);
        });
    };

    const renderFolderPicker = () => {
        return (
            <div role="dialog">
                <Dialog
                    hidden={!showFolderPicker}
                    title={t(ResourceKeys.modelRepository.types.local.folderPicker.dialog.title)}
                    modalProps={{
                        className: 'folder-picker-dialog',
                        isBlocking: false
                    }}
                    dialogContentProps={{
                        subText: currentFolder && t(ResourceKeys.modelRepository.types.local.folderPicker.dialog.subText, {folder: currentFolder})
                    }}
                    onDismiss={dismissFolderPicker}
                >
                    <div className="folder-links">
                        <DefaultButton
                            className="folder-button"
                            iconProps={{ iconName: NAVIGATE_BACK }}
                            text={t(ResourceKeys.modelRepository.types.local.folderPicker.command.navigateToParent)}
                            ariaLabel={t(ResourceKeys.modelRepository.types.local.folderPicker.command.navigateToParent)}
                            onClick={onNavigateBack}
                            disabled={currentFolder === getRootFolder()}
                        />
                        {showError ? <div className="no-folders-text">{t(ResourceKeys.modelRepository.types.local.folderPicker.dialog.error)}</div> :
                            subFolders && subFolders.length > 0 ?
                                subFolders.map(folder =>
                                    <DefaultButton
                                        className="folder-button"
                                        iconProps={{ iconName: FOLDER }}
                                        key={folder}
                                        text={folder}
                                        onClick={onClickFolderName(folder)}
                                    />)
                                :
                                <div className="no-folders-text">{t(ResourceKeys.modelRepository.types.local.folderPicker.dialog.noFolderFoundText)}</div>}
                    </div>
                    <DialogFooter>
                        <PrimaryButton onClick={onSelectFolder} text={t(ResourceKeys.modelRepository.types.local.folderPicker.command.select)} disabled={!currentFolder}/>
                        <DefaultButton onClick={dismissFolderPicker} text={t(ResourceKeys.modelRepository.types.local.folderPicker.command.cancel)} />
                    </DialogFooter>
                </Dialog>
            </div>
        );
    };

    return(
        <>
            <ListItemLocalLabel repoType={repoType}/>
            <TextField
                className="local-folder-textbox"
                label={t(ResourceKeys.modelRepository.types.local.textBoxLabel)}
                ariaLabel={t(ResourceKeys.modelRepository.types.local.textBoxLabel)}
                value={currentFolder}
                readOnly={false}
                errorMessage={errorKey ? t(errorKey) : ''}
                onChange={onFolderPathChange}
            />
            <DefaultButton
                className="local-folder-launch"
                text={t(ResourceKeys.modelRepository.types.local.folderPicker.command.openPicker)}
                ariaLabel={t(ResourceKeys.modelRepository.types.local.folderPicker.command.openPicker)}
                onClick={onShowFolderPicker}
            />
            {renderFolderPicker()}
        </>
    );
};
