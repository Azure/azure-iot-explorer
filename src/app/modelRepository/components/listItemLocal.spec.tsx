/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import 'jest';
import * as React from 'react';
import { mount, shallow } from 'enzyme';
import { act } from 'react-dom/test-utils';
import { Dialog } from '@fluentui/react';
import { ListItemLocal } from './listItemLocal';
import * as Utils from '../../shared/utils/utils';
import { REPOSITORY_LOCATION_TYPE } from '../../constants/repositoryLocationTypes';
import { getInitialModelRepositoryFormState } from '../state';
import { getInitialModelRepositoryFormOps } from '../interface';
import { ResourceKeys } from '../../../localization/resourceKeys';

describe('ListItemLocal', () => {
    it('matches snapshot for local folder', () => {
        const wrapper = shallow(
            <ListItemLocal
                index={0}
                item={{
                    repositoryLocationType: REPOSITORY_LOCATION_TYPE.Local,
                    value: 'c:/models'
                }}
                repoType={REPOSITORY_LOCATION_TYPE.Local}
                formState={[getInitialModelRepositoryFormState(), getInitialModelRepositoryFormOps()]}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });


    it('renders no folder text when no sub folder is retrieved', () => {
        jest.spyOn(Utils, 'getRootFolder').mockReturnValue('c:/models');

        const wrapper = mount(
            <ListItemLocal
                index={0}
                item={{
                    repositoryLocationType: REPOSITORY_LOCATION_TYPE.Local,
                    value: 'c:/models'
                }}
                repoType={REPOSITORY_LOCATION_TYPE.Local}
                formState={[getInitialModelRepositoryFormState(), getInitialModelRepositoryFormOps()]}
            />
        );

        act(() => wrapper.find('.local-folder-launch').first().props().onClick(undefined));
        wrapper.update();

        const dialog = wrapper.find(Dialog).first();
        expect(dialog.children().props().hidden).toBeFalsy();
        expect(dialog.children().props().children[0].props.children[0].props.disabled).toBeTruthy();
        expect(dialog.children().props().children[0].props.children[1].props.children).toEqual(ResourceKeys.modelRepository.types.local.folderPicker.dialog.noFolderFoundText);
    });

    it('renders folders when sub folders retrieved', () => {
        const subFolders = ['documents', 'pictures'];
        jest.spyOn(Utils, 'getRootFolder').mockReturnValue('d:/');

        const realUseState = React.useState;
        jest.spyOn(React, 'useState').mockImplementationOnce(() => realUseState(subFolders));

        const wrapper = mount(
            <ListItemLocal
                index={0}
                item={{
                    repositoryLocationType: REPOSITORY_LOCATION_TYPE.Local,
                    value: 'c:/models'
                }}
                repoType={REPOSITORY_LOCATION_TYPE.Local}
                formState={[getInitialModelRepositoryFormState(), getInitialModelRepositoryFormOps()]}
            />
        );

        act(() => wrapper.find('.local-folder-launch').first().props().onClick(null));
        wrapper.update();

        const dialog = wrapper.find(Dialog).first();
        expect(dialog.children().props().hidden).toBeFalsy();
        expect(dialog.children().props().children[0].props.children[0].props.text).toEqual(ResourceKeys.modelRepository.types.local.folderPicker.command.navigateToParent);
        expect(dialog.children().props().children[0].props.children[0].props.disabled).toBeFalsy();
        expect(dialog.children().props().children[0].props.children[1].length).toEqual(subFolders.length);
        expect(dialog.children().props().children[0].props.children[1][0].props.text).toEqual(subFolders[0]);
    });
});
