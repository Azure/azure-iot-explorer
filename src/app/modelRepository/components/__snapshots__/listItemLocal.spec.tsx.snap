// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemLocal matches snapshot for local folder 1`] = `
<Fragment>
  <ListItemLocalLabel
    repoType="LOCAL"
  />
  <StyledTextFieldBase
    ariaLabel="modelRepository.types.local.textBoxLabel"
    className="local-folder-textbox"
    errorMessage=""
    label="modelRepository.types.local.textBoxLabel"
    onChange={[Function]}
    readOnly={false}
    value="c:/models"
  />
  <CustomizedDefaultButton
    ariaLabel="modelRepository.types.local.folderPicker.command.openPicker"
    className="local-folder-launch"
    onClick={[Function]}
    text="modelRepository.types.local.folderPicker.command.openPicker"
  />
  <div
    role="dialog"
  >
    <Dialog
      dialogContentProps={
        Object {
          "subText": "modelRepository.types.local.folderPicker.dialog.subText",
        }
      }
      hidden={true}
      modalProps={
        Object {
          "className": "folder-picker-dialog",
          "isBlocking": false,
        }
      }
      onDismiss={[Function]}
      title="modelRepository.types.local.folderPicker.dialog.title"
    >
      <div
        className="folder-links"
      >
        <CustomizedDefaultButton
          ariaLabel="modelRepository.types.local.folderPicker.command.navigateToParent"
          className="folder-button"
          disabled={false}
          iconProps={
            Object {
              "iconName": "NavigateBack",
            }
          }
          onClick={[Function]}
          text="modelRepository.types.local.folderPicker.command.navigateToParent"
        />
        <div
          className="no-folders-text"
        >
          modelRepository.types.local.folderPicker.dialog.noFolderFoundText
        </div>
      </div>
      <StyledDialogFooterBase>
        <CustomizedPrimaryButton
          disabled={false}
          onClick={[Function]}
          text="modelRepository.types.local.folderPicker.command.select"
        />
        <CustomizedDefaultButton
          onClick={[Function]}
          text="modelRepository.types.local.folderPicker.command.cancel"
        />
      </StyledDialogFooterBase>
    </Dialog>
  </div>
</Fragment>
`;
